mapusaurus/hmda/views.py

```diff
@@ -18,48 +19,56 @@ def loan_originations(request):
     action_taken_param = request.GET.get('action_taken')
     lender_hierarchy = request.GET.get('lh')
     peers = request.GET.get('peers')
-    geos = get_censustract_geos(request)
-    institution_selected = Institution.objects.filter(pk=institution_id).first()
-    metro_selected = Geo.objects.filter(geo_type=Geo.METRO_TYPE, geoid=metro).first()
-    if action_taken_param:
-        action_taken_selected = action_taken_param.split(',')
-    else:
-        action_taken_selected = []
-    query = HMDARecord.objects.filter(base_hmda_query())
-    if action_taken_selected:
-        query = query.filter(action_taken__in=action_taken_selected)
-    if institution_selected:
-        if lender_hierarchy == 'true':
-            hierarchy_list = institution_selected.get_lender_hierarchy(False, False)
-            if len(hierarchy_list) > 0:
-                query = query.filter(institution__in=hierarchy_list) 
+    census_tracts = get_censustract_geos(request)
+
+    query = HMDARecord.objects.filter(base_hmda_query())            
+
+    #if lender param key is passed in
+    if institution_id:
+        institution_selected = get_object_or_404(Institution, pk=institution_id)
+        if institution_selected: 
```

Comment:
May not need this check as it will throw 404 after above check (get_object_or_404)